components/contribution-flow/StepDetails.js

```diff
@@ -220,7 +221,7 @@ const StepDetails = ({ onChange, data, collective, tier, showFeesOnTop, router }
           />
         </Box>
       )}
-      {temporaryInterval !== undefined && (
+      {temporaryInterval !== undefined && tier?.interval !== INTERVALS.flexible && (
```

Comment:
Since we're not calling `setTemporaryInterval` with flexible, is this check really needed?